Mage/src/main/java/mage/players/PlayerImpl.java

```diff
@@ -2124,20 +2124,12 @@ private int doDamage(int damage, UUID attackerId, Ability source, Game game, boo
         if (damage < 1) {
             return 0;
         }
-        if (!canDamage(game.getObject(attackerId), game)) {
-            MageObject sourceObject = game.getObject(attackerId);
-            game.informPlayers(damage + " damage "
-                    + (sourceObject == null ? "" : "from " + sourceObject.getLogName())
-                    + " to " + getLogName()
-                    + (damage > 1 ? " were" : "was") + " prevented because of protection");
-            return 0;
-        }
         DamageEvent event = new DamagePlayerEvent(playerId, attackerId, playerId, damage, preventable, combatDamage);
         event.setAppliedEffects(appliedEffects);
         if (game.replaceEvent(event)) {
             return 0;
         }
-        int actualDamage = event.getAmount();
+        int actualDamage = checkProtectionAbilities(event, attackerId, source, game);
```

Comment:
@xenohedron you are changed events order here. 

Old order: protection -> prevent (prevent effects will be ignored on active protection).
Now: prevent -> protection (prevent events applies all the time).

So it affect game logic in that fix. Is that ok or buggy?

Prevent effect that would be applies with protected permanents now:
![shot_230727_211159](https://github.com/magefree/mage/assets/8344157/e16c5292-6f00-400a-ae58-0d4a0f83eeaa)

Reply:
Does that work differently when redirecting damage from a protected permanent to another?
Because for PermanentImpl, this is definitively prevent then protection.

Reply:
New player damage logic is same as existing permanent damage logic, yeah.

Reply:
See #10680, I think both old and new are incorrect.